libusb/os/windows_hotplug.c

```diff
@@ -0,0 +1,307 @@
+#include "libusbi.h"
+#include "threads_windows.h"
+
+#include "windows_common.h"
+#include "windows_hotplug.h"
+
+#include <stdio.h>
+#include <dbt.h>
+#include <usbiodef.h>
+
+/* The Windows Hotplug system is a three steps process.
+ * 1. We create a monitor on GUID_DEVINTERFACE_USB_DEVICE via a hidden window.
+ * 2. Upon notification of an event, we run the current windows backend to get the list of devices.
+ *    This updates the hotplug status of each device to one of three values {UNCHANGED, ARRIVED, LEFT}.
+ * 3. According to the value, we generate events to libusb client via hotplug callbacks. */
+
+static HWND windows_event_hWnd;
+static HANDLE windows_event_thread_handle;
+static DWORD WINAPI windows_event_thread_main(LPVOID lpParam);
+static LRESULT CALLBACK windows_proc_callback(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam);
+
+#define log_error(operation) do { \
+	usbi_err(NULL, "%s failed with error: %s", operation, windows_error_str(0)); \
+} while (0)
+
+int windows_start_event_monitor(void)
+{
+	windows_event_thread_handle = CreateThread(
+		NULL,
+		0,
+		windows_event_thread_main,
+		NULL,
+		0,
+		NULL
+	);
+
+	if(windows_event_thread_handle == NULL)
+	{
```

Comment:
I notice this is a new code style with newline before opening braces, and tabs instead of spaces. But if() without space seems wrong to me, and you use spaces further down.

Reply:
OK for the missing space in between "if" and "(", sorry.
Regarding the code style of having opening curly brace on a new line: almost all Win32 code I saw is written in this way (example here: https://github.com/microsoft/Windows-classic-samples/blob/main/Samples/Win7Samples/begin/LearnWin32/HelloWorld/cpp/main.cpp) so I thought people familiar with it would feel more confortable. The same is true for the capitalisation of parameter, like `lParam` and other details. I try my best to keep commonly used Win32 names so that they are immediately recognizable, while mixing with names defined somewhere else in libsub and using underscores and all small caps, i.e. I wanted to avoid `l_param `and `UnrefList` ;-)
Regarding tabs instead of space, I didn't get it. Didn't we decide to use tabs?https://github.com/libusb/libusb/pull/1406#issuecomment-1872127406 ?

Anyway, I can adapt my code, no problem, just tell me.

Reply:
Yes, I am fine with using commonly used Win32 names. Just checking. Yes tabs is not wrong, just pointing out the chosen style. winusb_*.c use tabs as well, but with trailing opening braces. The darwin code uses spaces... This is the topic of #1442 so it will be settled there, I guess. Just use whatever for now and it will be reformatted if needed later.

Reply:
Yes, all the time spent discussing this minutia can be automated away with clang-format, allowing reviewers to focus on what really matters.